packages/components/src/tab-panel/index.tsx

```diff
@@ -92,112 +85,156 @@ const UnforwardedTabPanel = (
 	ref: ForwardedRef< any >
 ) => {
 	const instanceId = useInstanceId( TabPanel, 'tab-panel' );
-	const [ selected, setSelected ] = useState< string >();
 
-	const handleTabSelection = useCallback(
-		( tabKey: string ) => {
-			setSelected( tabKey );
-			onSelect?.( tabKey );
+	const prependInstanceId = useCallback(
+		( tabName: string | undefined ) => {
+			if ( typeof tabName === 'undefined' ) {
+				return;
+			}
+			return `${ instanceId }-${ tabName }`;
+		},
+		[ instanceId ]
+	);
+
+	const tabStore = Ariakit.useTabStore( {
```

Comment:
Just having a look at this PR — is there a reason why we imported `* from Ariakit` instead of importing only the needed components / functions?

Reply:
No, actually. Importing just the needed items would have been better. I'll plan a followup.

Reply:
Just in case this is a factor, besides code style/aesthetics, there's no difference between importing the namespace and components individually. In the docs, we sometimes use the namespace import. The reasons are described in this section of the docs: https://ariakit.org/guide/coding-guidelines#import-namespace